sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -1976,33 +2020,37 @@ pi_result cuda_piContextRelease(pi_context ctxt) {
 
   assert(ctxt != nullptr);
 
-  if (ctxt->decrement_reference_count() > 0) {
-    return PI_SUCCESS;
-  }
-  ctxt->invoke_extended_deleters();
+  if (ctxt->backend_has_ownership()) {
```

Comment:
For readability, I think this should be an early exit, similar to the first 3 lines below. However, I believe this cuts out too much of the function as we will still need to destroy the `_pi_context`, which is what the unique_ptr further down is responsible for.

Reply:
Okay, I have updated this to an early exit after the unique_ptr